src/transform/ReferenceCollection.js

```diff
@@ -0,0 +1,253 @@
+  Array.prototype.slice.call(refTextContainer.childNodes)
+    .filter(NodeUtilities.isNodeTypeElementOrText)
+    .forEach(cloneNodeIntoFragmentDiv)
+
+  const removalSelector = 'sup[id^=cite_ref], .mw-cite-backlink'
+  Polyfill.querySelectorAll(fragDiv, removalSelector)
+    .forEach(node => node.remove())
+
+  return fragDiv.innerHTML.trim()
+}
+
+/**
+ * Get closest element to node which has class `reference`. If node itself has class `reference`
+ * returns the node.
+ * @param  {!Node} sourceNode
+ * @return {?HTMLElement}
+ */
+const closestReferenceClassElement = sourceNode => {
+  if (Polyfill.matchesSelector(sourceNode, REFERENCE_SELECTOR)) {
+    return sourceNode
+  }
+  return ElementUtilities.findClosestAncestor(sourceNode, REFERENCE_SELECTOR)
+}
+
+/**
+ * Reference item model.
+ */
+class ReferenceItem {
+  /**
+   * ReferenceItem construtor.
+   * @param {!string} id
+   * @param {!DOMRect} rect
+   * @param {?string} text
+   * @param {?string} html
+   */
+  constructor(id, rect, text, html) {
+    this.id = id
+    this.rect = rect
+    this.text = text
+    this.html = html
+  }
+}
+
+/**
+ * Converts node to ReferenceItem.
+ * @param  {!Document} document
+ * @param  {!Node} node
+ * @return {!ReferenceItem}
+ */
+const referenceItemForNode = (document, node) => new ReferenceItem(
+  closestReferenceClassElement(node).id,
+  node.getBoundingClientRect(),
+  node.textContent,
+  collectRefText(document, node)
+)
+
+/**
+ * Container for nearby references including the index of the selected reference.
+ */
+class NearbyReferences {
+/**
+ * @param  {!number} selectedIndex
+ * @param  {!Array.<ReferenceItem>} referencesGroup
+ * @return {!NearbyReferences}
+ */
+  constructor(selectedIndex, referencesGroup) {
+    this.selectedIndex = selectedIndex
+    this.referencesGroup = referencesGroup
+  }
+}
+
+/**
+ * Closure around a node for getting previous or next sibling.
+ *
+ * @typedef SiblingGetter
+ * @param  {!Node}  node
+ * @return {?Node}
+ */
+
+/**
+  * Closure around `collectedNodes` for collecting reference nodes.
+  *
+  * @typedef Collector
+  * @param  {!Node}  node
+  * @return {void}
+  */
+
+/**
+ * Get adjacent non-whitespace node.
+ * @param  {!Node}  node
+ * @param  {!SiblingGetter} siblingGetter
+ * @return {?Node}
+ */
+const adjacentNonWhitespaceNode = (node, siblingGetter) => {
+  let currentNode = node
+  do {
+    currentNode = siblingGetter(currentNode)
+  } while (isWhitespaceTextNode(currentNode))
+  return currentNode
+}
+
+/**
+ * Collect adjacent reference nodes.
+ * @param  {!Node} node
+ * @param  {!SiblingGetter} siblingGetter
+ * @param  {!Collector} nodeCollector
+ * @return {void}
+ */
+const collectAdjacentReferenceNodes = (node, siblingGetter, nodeCollector) => {
+  let currentNode = node
+  while (true) {
+    currentNode = adjacentNonWhitespaceNode(currentNode, siblingGetter)
+    if (hasCitationLink(currentNode)) {
+      nodeCollector(currentNode)
+      continue
```

Comment:
If you invert this condition, you can replace the continue here with the break below.

Reply:
Doh! Good catch! Fixed